From a5ab86b5740cb2acaf7907d0b6ec095705509db2 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Wed, 3 Jul 2013 22:10:01 +0300
Subject: [PATCH 274/290] be: fix compilation errors

- s/M0_BTS_STABLE/M0_BTS_DONE/
- Comment out the definition of m0_be__tx_state_post().
---
 be/btree.h | 2 +-
 be/tx.c    | 3 +++
 be/ut/tx.c | 2 +-
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/be/btree.h b/be/btree.h
index cbf92ac..4ddfe6a 100644
--- a/be/btree.h
+++ b/be/btree.h
@@ -163,7 +163,7 @@ M0_INTERNAL void m0_be_btree_credit(const struct m0_be_btree *tree,
  * into the mmaped segment.
  *
  * Note2: When tx::bt_sm state transits from M0_BTS_PREPARE to M0_BTS_PLACED
- * or M0_BTS_STABLE then data passed to the function in @key and @value become
+ * or M0_BTS_DONE then data passed to the function in @key and @value become
  * persistent.
  *
  */
diff --git a/be/tx.c b/be/tx.c
index b422a85..3da8028 100644
--- a/be/tx.c
+++ b/be/tx.c
@@ -533,6 +533,8 @@ static void tx_fail(struct m0_be_tx *tx, int err)
 	M0_POST(m0_be__tx_invariant(tx));
 }
 
+#if 0 /* XXX <<<<<<<
+       * Do not remove this section till 2013-Jul-05. Thank you.  --vvv */
 static void
 _tx_state_set(struct m0_sm_group *grp M0_UNUSED, struct m0_sm_ast *ast)
 {
@@ -578,6 +580,7 @@ m0_be__tx_state_post(struct m0_be_tx *tx, enum m0_be_tx_state state)
 	m0_sm_ast_post(tx->t_sm.sm_grp, &tx->t_ast);
 	M0_LEAVE();
 }
+#endif /* XXX >>>>>>> */
 
 static struct m0_be_tx_engine *tx_engine(const struct m0_be_tx *tx)
 {
diff --git a/be/ut/tx.c b/be/ut/tx.c
index 0023406..f44268a 100644
--- a/be/ut/tx.c
+++ b/be/ut/tx.c
@@ -46,7 +46,7 @@ static void tx_test(struct m0_be *be)
 	m0_be_op_init(&op);
 	m0_be_tx_credit_init(&cred);
 
-	m0_be_tx_init(&tx, ++g_tid, be, &g_grp, noop, noop, NULL, NULL);
+	m0_be_tx_init(&tx, ++g_tid, be, &g_grp, noop, noop, true, NULL, NULL);
 	m0_be_allocator_credit(alloc, M0_BAO_ALLOC, sizeof *p, 0, &cred);
 
 	m0_sm_group_lock(&g_grp);
-- 
1.8.3.2

